operator/apis/loki/v1beta1/rulerconfig_types.go

```diff
@@ -118,6 +118,37 @@ type AlertManagerSpec struct {
 	// +kubebuilder:validation:Optional
 	// +operator-sdk:csv:customresourcedefinitions:type=spec,displayName="Alert Relabel Configuration"
 	RelabelConfigs []RelabelConfig `json:"relabelConfigs,omitempty"`
+
+	// Client configuration for reaching the alertmanager endpoint.
+	//
+	// +optional
+	// +kubebuilder:validation:Optional
+	// +operator-sdk:csv:customresourcedefinitions:type=spec,displayName="TLS Config"
+	Client *AlertManagerClientConfig `json:"client,omitempty"`
+}
+
+type AlertManagerClientConfig struct {
+	TLS        *AlertManagerClientTLSConfig  `json:"tls,omitempty"`
+	HeaderAuth *AlertManagerClientHeaderAuth `json:"headerAuth,omitempty"`
+	BasicAuth  *AlertManagerClientBasicAuth  `json:"basicAuth,omitempty"`
+}
+
+type AlertManagerClientBasicAuth struct {
+	Username *string `json:"username,omitempty"`
+	Password *string `json:"password,omitempty"`
+}
+
+type AlertManagerClientHeaderAuth struct {
+	Type            *string `json:"type,omitempty"`
+	Credentials     *string `json:"credentials,omitempty"`
+	CredentialsFile *string `json:"credentialsFile,omitempty"`
+}
+
+type AlertManagerClientTLSConfig struct {
```

Comment:
Why is our linter not complaining about the missing doc comments here? :confused: